Updating readme to include guide on integrating with docker agent (#1842)
DataDog/integrations-extras: twingate/README.md

2. You must deploy the Twingate Connector. To enable real-time connection logs, see the [Twingate documentation][3].

### Configure the Datadog Agent
#### Systemd Connector
1. Set up [Datadog journald integration][5].
2. Replace `journald.d/conf.yaml` with the following configuration:

>    ```

Reviewer: ```suggestion
   ```yaml
```

          pattern: "ANALYTICS "
   ```
3. Add the `dd-agent` user to the `systemd-journal` group by using `usermod -a -G systemd-journal dd-agent`.

> 4. Restart the Datadog Agent by running `service datadog-agent restart`

Reviewer: ```suggestion
4. Restart the Datadog Agent by running `service datadog-agent restart`.
```

> 5. Confirm that the Twingate Analytic log appears in the [Log Explorer][10]

Reviewer: ```suggestion
5. Confirm that the Twingate Analytic log appears in the [Log Explorer][10].
```

#### Docker Connector

> ##### (Host Agent) Set up Datadog Docker integration

Reviewer: ```suggestion
##### Set up Datadog Docker integration for the Host Agent
```

> - Adding the following lines to `datadog.yaml`:

Reviewer: ```suggestion
Add the following lines to the `datadog.yaml` configuration file:
```

> ```

Reviewer: ```suggestion
```yaml
```

logs_enabled: true
listeners:
- name: docker
config_providers:
- name: docker
polling: true
logs_config:
container_collect_all: true
container_exclude: ["image:.*"]
container_include: ["image:twingate/connector"]
```

> - Add the `dd-agent` user to the `docker` group by using `usermod -a -G docker dd-agent`

Reviewer: ```suggestion
- Add the `dd-agent` user to the `docker` group by using `usermod -a -G docker dd-agent`.
```

> - Restart the Datadog Agent by running `service datadog-agent restart`

Reviewer: ```suggestion
- Restart the Datadog Agent by running `service datadog-agent restart`.
```

> ##### (Container Agent) Set up Datadog Docker integration

Reviewer: ```suggestion
##### Set up Datadog Docker integration for the Container Agent
```

- add additional parameters `-e DD_CONTAINER_EXCLUDE="image:.*"` and `-e DD_CONTAINER_INCLUDE="image:twingate/connector"` in the docker run command

> ```

Reviewer: ```suggestion
```shell
```

docker run -d --name datadog-agent \
           --cgroupns host \
           --pid host \
           -e DD_API_KEY=xxx \
           -e DD_LOGS_ENABLED=true \
           -e DD_LOGS_CONFIG_CONTAINER_COLLECT_ALL=true \
           -e DD_CONTAINER_EXCLUDE="image:.*" \
           -e DD_CONTAINER_INCLUDE="image:twingate/connector" \
           -v /var/run/docker.sock:/var/run/docker.sock:ro \
           -v /var/lib/docker/containers:/var/lib/docker/containers:ro \
           -v /proc/:/host/proc/:ro \
           -v /opt/datadog-agent/run:/opt/datadog-agent/run:rw \
           -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
           gcr.io/datadoghq/agent:latest
```

##### Set up Twingate Connector with additional docker parameters

> Adding additional label `com.datadoghq.ad.logs` to the Twingate Connector docker run command:

Reviewer: ```suggestion
Add the label `com.datadoghq.ad.logs` to the Twingate Connector docker run command:
```

> ```

Reviewer: ```suggestion
```shell
```

docker run -d --sysctl net.ipv4.ping_group_range="0 2147483647" \
  -l "com.datadoghq.ad.logs"='[{"service":"Twingate Connection","source":"Twingate","log_processing_rules":[{"type":"include_at_match","name":"analytics","pattern":"ANALYTICS"},{"type":"mask_sequences","name":"remove_analytics","replace_placeholder":"","pattern":"ANALYTICS "}]}]' \
  --env TENANT_URL="https://xxx.twingate.com" \
  --env ACCESS_TOKEN="xxx" \
  --env REFRESH_TOKEN="xxx" \
  --env TWINGATE_LABEL_HOSTNAME="`hostname`" \
  --name "twingate-golden-seal" \
  --restart=unless-stopped \
  $(docker run --help | grep -- --pull >/dev/null && echo "--pull=always") twingate/connector:1
```

> Note: The Twingate Connector container needs to be recreated to add the additional label 

Reviewer: ```suggestion
**Note**: The Twingate Connector container needs to be recreated to add the new label 
```

[10]: https://app.datadoghq.com/logs
[11]: https://docs.datadoghq.com/getting_started/agent/
[12]: https://app.datadoghq.com/dashboard/lists
[13]: https://docs.datadoghq.com/containers/docker/log/?tab=containerinstallation#installation
[14]: https://docs.datadoghq.com/agent/guide/autodiscovery-management/?tab=agent_
[15]: https://docs.datadoghq.com/containers/docker/log/?tab=containerinstallation#log-integrations

> [16]: https://gist.github.com/chenbishop/3f1b9833f274716949a40af385e2f5c2

Reviewer: ```suggestion
```

> [17]: https://gist.github.com/chenbishop/7df60e074e7e856e17d867b31f6e95a1

Reviewer: ```suggestion
```

> [18]: https://gist.github.com/chenbishop/3be2bcc90d7d77288a843de469f0b3d0

Reviewer: ```suggestion
```

> [19]: https://gist.github.com/chenbishop/a5405cc00bc1ac50fdbd86ef4ad23599

Reviewer: ```suggestion
```
